src/main/java/net/openhft/hashing/WyHash.java

```diff
@@ -0,0 +1,316 @@
+                (byte)Primitives.unsignedByte(hi),
+                (byte)Primitives.unsignedByte(hi >> 8),
+                (byte)Primitives.unsignedByte(hi >> 16),
+                (byte)Primitives.unsignedByte(hi >> 24),
+                (byte)Primitives.unsignedByte(lo),
+                (byte)Primitives.unsignedByte(lo >> 8),
+                (byte)Primitives.unsignedByte(lo >> 16),
+                (byte)Primitives.unsignedByte(lo >> 24)
+            };
+            return WyHash.INSTANCE.wyHash64(seed(), bytes,
+                                            UnsafeAccess.INSTANCE, BYTE_BASE, bytes.length);
+        }
+
+        @Override
+        public long hashInt(int input) {
+            input = NATIVE_WY.toLittleEndian(input);
+            byte[] bytes = new byte[] {
+                (byte)Primitives.unsignedByte(input),
+                (byte)Primitives.unsignedByte(input >> 8),
+                (byte)Primitives.unsignedByte(input >> 16),
+                (byte)Primitives.unsignedByte(input >> 24)
+            };
+            return WyHash.INSTANCE.wyHash64(seed(), bytes,
+                                            UnsafeAccess.INSTANCE, BYTE_BASE, bytes.length);
+        }
+
+        @Override
+        public long hashShort(short input) {
+            input = NATIVE_WY.toLittleEndian(input);
+            byte[] hash = new byte[] {
+                (byte)Primitives.unsignedByte(input),
+                (byte)Primitives.unsignedByte(input >> 8)
+            };
+            return WyHash.INSTANCE.wyHash64(seed(), hash,
+                                            UnsafeAccess.INSTANCE, BYTE_BASE, hash.length);
+        }
+
+        @Override
+        public long hashChar(final char input) {
+            return hashShort((short)input);
+        }
+
+        @Override
+        public long hashByte(final byte input) {
+            byte[] bytes = new byte[]{input};
+            return WyHash.INSTANCE.wyHash64(seed(), bytes,
+                                            UnsafeAccess.INSTANCE, BYTE_BASE, bytes.length);
+        }
+
+        @Override
+        public long hashVoid() {
+            return 0;
+        }
+
+        @Override
+        public <T> long hash(final T input, final Access<T> access,
+                             final long off, final long len) {
+            long seed = seed();
+            if (access.byteOrder(input) == LITTLE_ENDIAN) {
+                return WyHash.INSTANCE.wyHash64(seed, input, access, off, len);
+            } else {
+                return BigEndian.INSTANCE.wyHash64(seed, input, access, off, len);
+            }
+        }
+    }
+    static LongHashFunction asLongHashFunctionWithSeed(long seed) {
+        return new AsLongHashFunctionSeeded(seed);
+    }
+
+    private static class AsLongHashFunctionSeeded extends AsLongHashFunction {
+        private static final long serialVersionUID = 0L;
+
+        private final long seed;
+        private final transient long voidHash = 0L;
```

Comment:
This field should be removed. BTW is it sure that wyhash's output for a zero-length sequence is 0 regardless of the seed?

Reply:
That is the behavior of the original C implementation, so matching the implementation output requires returning 0 for 0 input.
https://github.com/wangyi-fudan/wyhash/blob/master/wyhash.h#L35